docs/docs/how_to/pydantic_compatibility.md

```diff
@@ -1,27 +1,102 @@
+- Pydantic v2 was released in June, 2023 (https://docs.pydantic.dev/2.0/blog/pydantic-v2-final/).
+- v2 contains has a number of breaking changes (https://docs.pydantic.dev/2.0/migration/).
 
-## LangChain Pydantic migration plan
+As of `langchain>=0.0.267`, LangChain allows users to install either Pydantic V1 or V2.
 
-As of `langchain>=0.0.267`, LangChain will allow users to install either Pydantic V1 or V2. 
-   * Internally LangChain will continue to [use V1](https://docs.pydantic.dev/latest/migration/#continue-using-pydantic-v1-features).
-   * During this time, users can pin their pydantic version to v1 to avoid breaking changes, or start a partial
-   migration using pydantic v2 throughout their code, but avoiding mixing v1 and v2 code for LangChain (see below).
+Internally, LangChain continues to use the [Pydantic V1](https://docs.pydantic.dev/latest/migration/#continue-using-pydantic-v1-features) via
+the v1 namespace of Pydantic 2.
 
-User can either pin to pydantic v1, and upgrade their code in one go once LangChain has migrated to v2 internally, or they can start a partial migration to v2, but must avoid mixing v1 and v2 code for LangChain.
+Because Pydantic does not support mixing .v1 and .v2 objects, users should be aware of a number of issues
+when using LangChain with Pydantic.
+
+:::note
+
+Pydantic 1 End of Life was in June 2024. LangChain will be dropping support for Pydantic 1 in the near future, 
+and likely migrating internally to Pydantic 2. The timeline is tentatively September.
+
+:::
+
+## 1. Passing Pydantic objects to LangChain APIs
+
+Most LangChain APIs that accept Pydantic objects have been updated to accept both Pydantic v1 and v2 objects.
+
+* Pydantic v1 objects correspond to subclasses of `pydantic.BaseModel` if `pydantic 1` is installed or subclasses of `pydantic.v1.BaseModel` if `pydantic 2` is installed.
+* Pydantic v2 objects correspond to subclasses of `pydantic.BaseModel` if `pydantic 2` is installed.
+
+
+| API                                    | Pydantic 1 | Pydantic 2                                                     |
+|----------------------------------------|------------|----------------------------------------------------------------|
+| `BaseChatModel.bind_tools`             | Yes        | langchain-core>=0.2.23, appropriate version of partner package |
+| `BaseChatModel.with_structured_output` | Yes        | langchain-core>=0.2.23, appropriate version of partner package |
+| `Tool.from_function`                   | Yes        | langchain-core>=0.2.23                                         |
+| `StructuredTool.from_function`         | Yes        | langchain-core>=0.2.23                                         |
+
+
+Partner packages that accept pydantic v2 objects via `bind_tools` or `with_structured_output` APIs:
+
+| Package Name        | pydantic v1 | pydantic v2 |
+|---------------------|-------------|-------------|
```

Comment:
what if we put langchain-core in here, and instead of the above table just have a list of the APIs and say: these all support pydantic 2 as of the following package versions:

Reply:
I was trying to guard against force install of incompatible package versions

Reply:
watcha mean? which versions would be incompatible? these are all min bounds

Reply:
Pip force install unless it's no longer possible